src/Bicep.Core.IntegrationTests/RegistryProviderTests.cs

```diff
@@ -58,12 +59,71 @@ public async Task Providers_published_to_a_registry_can_be_compiled()
 output joke string = dadJoke.body.joke
 """);
 
-        // TODO uncomment the below once the 3rd party logic has been implemented
-        // result.Should().NotHaveAnyDiagnostics();
-        // result.Template.Should().NotBeNull();
+        result.Should().NotHaveAnyDiagnostics();
+        result.Template.Should().NotBeNull();
+    }
+
+    [TestMethod]
+    public async Task Third_party_namespace_errors_with_configuration()
+    {
+        // types taken from https://github.com/Azure/bicep-registry-providers/tree/21aadf24cd6e8c9c5da2db0d1438df9def548b09/providers/http
+        var fileSystem = FileHelper.CreateMockFileSystemForEmbeddedFiles(
+            typeof(RegistryProviderTests).Assembly,
+            "Files/RegistryProviderTests/HttpProvider");
+
+        var registry = "example.azurecr.io";
+        var repository = $"test/provider/http";
+
+        var services = GetServiceBuilder(fileSystem, registry, repository);
+
+        await DataSetsExtensions.PublishProviderToRegistryAsync(services.Build(), "/types/index.json", $"br:{registry}/{repository}:1.2.3");
+
+        var result = await CompilationHelper.RestoreAndCompile(services, """
+provider 'br:example.azurecr.io/test/provider/http@1.2.3' with {{}}
```

Comment:
For the raw literal syntax, you don't need to escape `{` and `}`. Here you're declaring a bicep file with `{{` and `}}`, which I don't think is the intention.

```suggestion
provider 'br:example.azurecr.io/test/provider/http@1.2.3' with {}
```